### PR TITLE
When must change password only show signout button in header

### LINKED ISSUE #11554: New user recommended setting password question
- Gitea version (or commit ref): 1.12
- Git version: 2.26.2
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
The newly created user checked "recommended password setting". However, after the user has changed the password when logging in, he/she jumps to the event page and cannot open it. The diagram below:
the demo url like this:https://domain.com/user/events
dont tell is session??
@zeripath @GiteaBot 
...

## Screenshots
image

### PR DESCRIPTION
**Pull request by zeripath**
When "Must Change Password" simplify the navbar header to only show the Signout button as all other links will redirect back in any case. 

This prevents the notifications icon from showing preventing initialization of the event-source and hence preventing redirect_to being set, however in addition do not set the `redirect_to` cookie if we are looking at the `/user/events` page.

Fix #11554

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]
This should go to the changelog. The navbar is a likely template users will customize; if I've got this right, this will break those customizations (or rather their customizations will lack this needed change).

**Comment by zeripath** [comment]
Make lg-tm work

File: templates/base/head_navbar.tmpl [hunk 1]
```diff
@@ -8,7 +8,9 @@
 		</div>
 	</div>
 
-	{{if .IsSigned}}
+	{{if and .IsSigned .MustChangePassword}}
+		{{/* No links */}}
+	{{else if .IsSigned}}
```
Comment (lafriks) [inline templates/base/head_navbar.tmpl:13]: It could be just simplified to `if and .IsSigned (not .MustChangePassword)` to not need empty if

Comment (zeripath) [reply templates/base/head_navbar.tmpl:13]: I considered that and originally wrote that but thought it better to be explicit and for the if statement to match the next one. We'd only have to write a two line comment to explain.

**Pull request merged**